test/suites/examples/install-karpenter-eks.yaml

```diff
@@ -0,0 +1,90 @@
+
+      aws eks update-kubeconfig --name $(params.test-cluster-name) --region ${AWS_REGION}
+
+      TEMPOUT=$(mktemp)
+      curl -fsSL "https://karpenter.sh/"$(params.karpenter-version)"/getting-started/getting-started-with-eksctl/cloudformation.yaml"  > $TEMPOUT \
+      && aws cloudformation deploy \
+        --stack-name "Karpenter-$(params.test-cluster-name)" \
+        --template-file "${TEMPOUT}" \
+        --capabilities CAPABILITY_NAMED_IAM \
+        --parameter-overrides "ClusterName=$(params.test-cluster-name)"
+
+      ROLE="    - rolearn: arn:aws:iam::$(params.account-id):role/KarpenterNodeRole-$(params.test-cluster-name)\n      username: system:node:{{EC2PrivateDNSName}}\n      groups:\n      - system:nodes\n      - system:bootstrappers"
+      kubectl get -n kube-system configmap/aws-auth -o yaml | awk "/mapRoles: \|/{print;print \"$ROLE\";next}1" > /tmp/aws-auth-patch.yml
+      kubectl patch configmap/aws-auth -n kube-system --patch "$(cat /tmp/aws-auth-patch.yml)"
+
+  - name: irsa
+    image: docker.io/weaveworks/eksctl:v0.86.0
+    script: |
+      eksctl create iamserviceaccount \
+        --cluster "$(params.test-cluster-name)" --name karpenter --namespace karpenter \
+        --role-name "$(params.test-cluster-name)-karpenter" \
+        --attach-policy-arn "arn:aws:iam::$(params.account-id):policy/KarpenterControllerPolicy-$(params.test-cluster-name)" \
+        --role-only \
+        --approve
+
+  - name: spot-slr
+    image: amazon/aws-cli
+    script: |
+      aws iam create-service-linked-role --aws-service-name spot.amazonaws.com || true
+
+  - name: helm-install-karpenter
+    image: alpine/helm:latest
+    script: |
+      aws eks update-kubeconfig --name $(params.test-cluster-name) --region ${AWS_REGION}
+
+      export KARPENTER_IAM_ROLE_ARN="arn:aws:iam::$(params.account-id):role/$(params.test-cluster-name)-karpenter"
+      export CLUSTER_ENDPOINT="$(aws eks describe-cluster --name $(params.test-cluster-name) --query "cluster.endpoint" --output text)"
+
+      helm repo add karpenter https://charts.karpenter.sh/
+      helm repo update
+
+      helm upgrade --install --namespace karpenter --create-namespace \
+        karpenter karpenter/karpenter \
+        --version $(params.karpenter-version) \
+        --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
+        --set clusterName=$(params.test-cluster-name) \
+        --set clusterEndpoint=${CLUSTER_ENDPOINT} \
+        --set aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-$(params.test-cluster-name) \
+        --wait
+
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: install-karpenter-eks-pipeline
+  namespace: karpenter-tests
+spec:
+  tasks:
+  - name: install-karpenter
+    taskRef:
+      name: install-karpenter-eks-task
+    params:
+    - name: test-cluster-name
+      value: "testing-ci-cluster"
+    - name: karpenter-version
+      value: "v0.13.1"
```

Comment:
This is gonna require some helm funkiness, since help doesn't support `v` as a prefix.